source/frontend/src/features/properties/map/acquisition/add/AcquisitionProperties.tsx

```diff
@@ -32,13 +34,19 @@ export const AcquisitionProperties: React.FunctionComponent<
               <Col>
                 <MapSelectorContainer
                   addSelectedProperties={(newProperties: IMapProperty[]) => {
-                    newProperties.forEach((property, index) => {
-                      const formProperty = PropertyForm.fromMapProperty(property);
-                      if (values.properties?.length === 0 && index === 0) {
-                        formikProps.setFieldValue(`region`, formProperty.region);
-                      }
-                      push(formProperty);
-                    });
+                    newProperties.reduce(async (promise, property) => {
+                      return promise.then(async () => {
+                        const formProperty = PropertyForm.fromMapProperty(property);
+                        if (property.pid) {
+                          const bcaSummary = await getPrimaryAddressByPid(property.pid);
+                          formProperty.address = bcaSummary?.address
+                            ? AddressForm.fromBcaAddress(bcaSummary?.address)
+                            : undefined;
+                          formProperty.legalDescription = bcaSummary?.legalDescription?.LEGAL_TEXT;
+                        }
+                        push(formProperty);
+                      });
+                    }, Promise.resolve());
```

Comment:
The new code seems to have discarded some previous functionality to set the region based on the first added property... could you please check?

```tsx
  if (values.properties?.length === 0 && index === 0) {
    formikProps.setFieldValue(`region`, formProperty.region);
  }
```

Reply:
hmm, I'll add this back.